js/src/forum/components/IndexPage.js

```diff
@@ -33,26 +33,27 @@ export default class IndexPage extends Page {
     // probably want to refresh the results. We will clear the discussion list
     // cache so that results are reloaded.
     if (app.previous instanceof IndexPage) {
-      app.cache.discussionList = null;
+      app.cache.discussionList.clear();
     }
 
     const params = this.params();
 
-    if (app.cache.discussionList) {
+    if (app.cache.discussionList.hasDiscussions()) {
       // Compare the requested parameters (sort, search query) to the ones that
       // are currently present in the cached discussion list. If they differ, we
       // will clear the cache and set up a new discussion list component with
       // the new parameters.
       Object.keys(params).some((key) => {
-        if (app.cache.discussionList.props.params[key] !== params[key]) {
-          app.cache.discussionList = null;
+        if (app.cache.discussionList.params[key] !== params[key]) {
+          app.cache.discussionList.clear();
           return true;
         }
       });
     }
 
-    if (!app.cache.discussionList) {
-      app.cache.discussionList = new DiscussionList({ params });
+    if (!app.cache.discussionList.hasDiscussions()) {
```

Comment:
Problem here - if the forum has no discussions, it will keep refreshing it every time the init method runs. That could potentially be an issue somewhere else ?

Reply:
I see this as being more of a positive though: every time someone goes to the index page, it checks if new discussions have been added

Reply:
This is a functionality change though. Not sure if we want that .

Reply:
I don't see any negatives to it. We'd need a bit of a hackier system to restore previous functionality, but that doesn't really bring any benefits: very rarely will forums have no discussions at all, and in the small case that they do, making it easier to see that new discussions have been added is a positive

Reply:
If there's no significant objections, can we resolve this for now?

Reply:
Sure, if you've tested it extensively.